numpy/ma/core.py

```diff
@@ -5868,6 +5868,7 @@ def __deepcopy__(self, memo=None):
         memo[id(self)] = copied
         for (k, v) in self.__dict__.items():
             copied.__dict__[k] = deepcopy(v, memo)
+        copied._sharedmask = False
```

Comment:
Will do the same here, will also be covered by refcount + owndata check, even if silly as hell.